inference-engine/src/multi_device/multi_device_plugin.cpp

```diff
@@ -219,34 +219,50 @@ IExecutableNetworkInternal::Ptr MultiDeviceInferencePlugin::LoadNetworkImpl(cons
     bool workModeAuto = workMode != fullConfig.end() && workMode->second == InferenceEngine::PluginConfigParams::YES;
     auto priorities = fullConfig.find(MultiDeviceConfigParams::KEY_MULTI_DEVICE_PRIORITIES);
 
-    // not found device priorities for -d AUTO use case
-    if (priorities == fullConfig.end()) {
-        if (workModeAuto) {
-            std::string allDevices;
-            auto availableDevices = GetCore()->GetAvailableDevices();
-            if (availableDevices.empty()) {
-                IE_THROW(NotFound) << "No available device found";
-            }
-            for (auto&& device : availableDevices) {
-                allDevices += device;
-                allDevices += ((device == availableDevices[availableDevices.size()-1]) ? "" : ",");
-            }
-            metaDevices = ParseMetaDevices(allDevices, fullConfig);
-            multiNetworkConfig.insert({MultiDeviceConfigParams::KEY_MULTI_DEVICE_PRIORITIES, allDevices});
-        } else {
-            IE_THROW() << "KEY_MULTI_DEVICE_PRIORITIES key is not set for " << GetName() << " device";
+    // if workMode is AUTO
+    if (workModeAuto) {
+        // check the configure and check if need to set PerfCounters configure to device
+        // and set filter configure
+        bool needPerfCounters = false;
+        std::map<std::string, std::string> filterConfig;
+        CheckConfig(fullConfig, needPerfCounters, filterConfig);
+        // filter the device that supports filter configure
+        auto strDevices = GetDeviceList(fullConfig);
+        auto metaDevices = ParseMetaDevices(strDevices, fullConfig);
+        auto supportDevices = FilterDevice(metaDevices, filterConfig);
+        if (supportDevices.size() == 0) {
+             IE_THROW() << "there is no device support the configure";
+        }
+        // replace the configure with configure that auto want to pass to device
+        // and reset the strDevices to support devices
+        std::vector<std::string> validConfigKey;
+        validConfigKey.push_back(PluginConfigParams::KEY_PERF_COUNT);
+        validConfigKey.push_back(PluginConfigParams::KEY_EXCLUSIVE_ASYNC_REQUESTS);
```

Comment:
minor comment: I assume this ASYNC_REQUESTS config key is only needed becuase OV has legacy impl and tests for that. but not sure of new devices like VPUX will support that, as the mode that is triggered by this key is obslolete.

So let's check that this checking this (evetually useless) config capability doesn't prevent us form using new devices like CUDA and KMB, and if this is the case- will need to remove maybe in the next PR